pkg/xray/watches.go

```diff
@@ -162,24 +201,77 @@ func packProjectResources(resources WatchProjectResources) []interface{} {
 		if len(res.RepoType) > 0 {
 			resourceMap["repo_type"] = res.RepoType
 		}
-		resourceMap["filter"] = packFilters(res.Filters)
+
+		resourceMap, errors := packFilters(res.Filters, resourceMap)
+		tflog.Error(ctx, fmt.Sprintf(`failed to pack filters: %v`, errors))
+
 		list = append(list, resourceMap)
 	}
 
 	return list
 }
 
-func packFilters(filters []WatchFilter) []interface{} {
-	var l []interface{}
-	for _, f := range filters {
-		m := map[string]interface{}{
-			"type":  f.Type,
-			"value": f.Value,
+type PackFilterFunc func(filter WatchFilter) (map[string]interface{}, error)
+
+func packStringFilter(filter WatchFilter) (map[string]interface{}, error) {
+	var value string
+	err := json.Unmarshal(filter.Value, &value)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
+		"type":  filter.Type,
+		"value": value,
+	}, nil
+}
+
+func packAntFilter(filter WatchFilter) (map[string]interface{}, error) {
+	var value WatchFilterAntValue
+	err := json.Unmarshal(filter.Value, &value)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
```

Comment:
You know, technically anyone who uses this function should be checking the err, so in theory, this is perfectly legit:

```go
err := json.Unmarshal(filter.Value, &value)
return map[string]interface{} {
    "exclude_patterns" : value.ExcludePatterns,
    "include_patterns" :  value.IncludePatterns
}, err
```
So, you always return both without condition - if err is ` != nil` then you get that here for free and the values in the map you don't care about (and neither should the caller)